lib/HLSL/HLModule.cpp

```diff
@@ -980,8 +980,13 @@ void HLModule::GetParameterRowsAndCols(Type *Ty, unsigned &rows, unsigned &cols,
   rows *= arraySize;
 }
 
-static Value *MergeGEP(GEPOperator *SrcGEP, GetElementPtrInst *GEP) {
-  IRBuilder<> Builder(GEP);
+static Value *MergeGEP(GEPOperator *SrcGEP, GEPOperator *GEP) {
+  IRBuilder<> Builder(GEP->getContext());
```

Comment:
Constant GEP merges were broken, which could be exposed in particular by the case where you have `constant GEP -> constant bitcast-to-base -> constant GEP`, and the middle bitcast-to-base is replaced with constant GEP, leading all to be merged together into one constant GEP.

Reply:
Good.  Actually, I realize that this case doesn't even require the first constant GEP, just the `constant bitcast -> constant GEP`, since bitcast gets replaced with constant GEP, then we need to merge the original constant GEP with the new one.